docs/TASK.md

```diff
@@ -48,87 +48,102 @@ I am a user who wants to define my custom task and bring my model to Disco, with
-> In a similar fashion, tasks expecting to read from browser memory (such as IndexedDB or local storage) shall be defined in `discojs-web` instead: `discojs/discojs-web/src/tasks/my_new_task.ts`
+For your custom model, the JSON model architecture is necessary, but the .bin weight file is optional : if you include the weights file, your model will be loaded with the passed weights. If a weights file is not specified, the weights for the model will be initialized randomly.
 
-### Model
+For more detail about how to define a `Task` and a `tf.LayersModel` for your own `TaskProvider`, continue reading.
 
-Start by creating a function in `my_new_task.ts` called `model` that returns the `tf.LayersModel`. In this example we programatically 
-define our model.
 
-After exporting the Task, you need to also export a function called `model` that returns the layers model. If you use a 
-pre-trained model, you can simply load and return said model in the function via `tf.loadLayersModel(modelPath)`.
+### Model
 
-> Note, it's important to add the `export` tag to the model as well as to the subsequent objects that we define.
+The interface let you load your model however you want, as long as you return a `tf.LayersModel` at the end. If you use a 
+pre-trained model, you can simply load and return said model in the function via `tf.loadLayersModel(modelPath)`.
 
 ```js
-export function model (_: string): tf.LayersModel {
+async function getModel (_: string): Promise<tf.LayersModel> {
   // Init model
   const model = tf.sequential()
 
   // Add layers
   model.add(...)
   
   return model
-
 ```
 
 Alternatively we can also load a pre-existing model; if we only provide a `model.json` file, then only the architecture of the model will be 
 loaded. If however in the same path we also include `weights.bin`, then pre-trained weights stored in these files will also be loaded to the model.
 
 ```js
-export async function model (modelPath: string): Promise<tf.LayersModel> {
+async function getModel (modelPath: string): Promise<tf.LayersModel> {
   return await tf.loadLayersModel(`file://${modelPath}`)
 }
 ```
 
-The models are stored in `disco/server/models/`, and it is also in the server side that we let disco know where exactly they are saved. In particular,
-if we look at `server/src/tasks.ts`, it is done as follows for the simple face example task (for our custom task we simply need to set
-our task with the model path).
-
-```js
-const simpleFaceModelPath = path.join(CONFIG.modelsDir, 'mobileNetV2_35_alpha_2_classes', 'model.json')
-
-// TODO, to add a custom model for a task, add the path here
-const MODEL_PATH = Map<string, string>()
-  .set(defaultTasks.simple_face.task.taskID, simpleFaceModelPath)
-
-```
+At runtime, the models are stored in `disco/server/models/`, and it is also in the server side that we let disco know where exactly they are saved. In particular,
```

Comment:
do you mean the initial models for each task, or even the freshly trained ones? (i guess it's true for FL but not for decentralized?)